src/azurelinux/3.0/cross/amd64-net9.0/Dockerfile

```diff
@@ -3,48 +3,47 @@ ARG ROOTFS_DIR=/crossrootfs/x64
 
 RUN /scripts/eng/common/cross/build-rootfs.sh x64 xenial --skipunmount
 
-# Build compiler-rt profile library for PGO instrumentation
-RUN mkdir compiler-rt_build && cd compiler-rt_build && \
-    BUILD_FLAGS="-v --sysroot=$ROOTFS_DIR" \
-    TARGET_TRIPLE="x86_64-linux-gnu" && \
-    cmake ../llvm-project.src/compiler-rt \
-        -DCOMPILER_RT_BUILD_PROFILE=ON \
-        -DCOMPILER_RT_BUILD_SANITIZERS=ON \
-        -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON \
-        \
-        -DCOMPILER_RT_BUILD_BUILTINS=OFF \
-        -DCOMPILER_RT_BUILD_CRT=OFF \
-        -DCOMPILER_RT_BUILD_GWP_ASAN=OFF \
-        -DCOMPILER_RT_BUILD_LIBFUZZER=OFF \
-        -DCOMPILER_RT_BUILD_MEMPROF=OFF \
-        -DCOMPILER_RT_BUILD_ORC=OFF \
-        -DCOMPILER_RT_BUILD_XRAY=OFF \
-        \
+ARG TARGET_TRIPLE="x86_64-linux-gnu"
+
+RUN cmake -S llvm-project.src/runtimes -B runtimes \
         -DCMAKE_BUILD_TYPE=Release \
         -DCMAKE_ASM_COMPILER=clang \
         -DCMAKE_C_COMPILER=clang \
         -DCMAKE_CXX_COMPILER=clang++ \
-        -DCMAKE_SHARED_LINKER_FLAGS="-fuse-ld=lld" \
+        -DCMAKE_ASM_COMPILER_TARGET="$TARGET_TRIPLE" \
+        -DCMAKE_C_COMPILER_TARGET="$TARGET_TRIPLE" \
+        -DCMAKE_CXX_COMPILER_TARGET="$TARGET_TRIPLE" \
+        # We're going to link the static libraries we build here into PIC images, so build the static libraries as PIC.
+        -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+        -DCMAKE_SYSROOT="$ROOTFS_DIR" \
+        # Specify linker to use for exes directly for CMake toolchain detection
         -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=lld" \
-        -DCMAKE_ASM_COMPILER_TARGET="${TARGET_TRIPLE}" \
-        -DCMAKE_C_COMPILER_TARGET="${TARGET_TRIPLE}" \
-        -DCMAKE_CXX_COMPILER_TARGET="${TARGET_TRIPLE}" \
-        -DCMAKE_ASM_FLAGS="${BUILD_FLAGS}" \
-        -DCMAKE_C_FLAGS="${BUILD_FLAGS}" \
-        -DCMAKE_CXX_FLAGS="${BUILD_FLAGS}" && \
-    make -j $(getconf _NPROCESSORS_ONLN)
-
-RUN LLVM_VERSION=18.1.3 LLVM_VERSION_MAJOR="${LLVM_VERSION%%.*}" SANITIZER_RUNTIMES_DIR=/sanitizer-runtimes && \
-    mkdir -p $ROOTFS_DIR/usr/lib/llvm-${LLVM_VERSION_MAJOR}/lib/clang/${LLVM_VERSION}/lib/linux/ && \
-    cp compiler-rt_build/lib/linux/libclang_rt.profile-x86_64.a $ROOTFS_DIR/usr/lib/llvm-${LLVM_VERSION_MAJOR}/lib/clang/${LLVM_VERSION}/lib/linux/ && \
-    mkdir ${SANITIZER_RUNTIMES_DIR} && \
-    cp compiler-rt_build/lib/linux/libclang_rt.*-x86_64.a $SANITIZER_RUNTIMES_DIR
-
+        # Don't search for tools in the sysroot as we're cross-compiling
+        -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM="NEVER" \
+        -DLLVM_USE_LINKER=lld \
```

Comment:
`CMAKE_EXE_LINKER_FLAGS` is needed for the `try_compile` commands in LLVM's configuration (otherwise the build tries to use the `ld` linker in the rootfs.

If we built Clang such that it defaulted to `lld`, then we could remove it (but then it would default for our builds as well, which is different than usual clang default behavior).